sdk/src/azure/core/az_http_response.c

```diff
@@ -158,6 +163,12 @@ AZ_NODISCARD az_result az_http_response_get_next_header(
     }
   }
 
+  if (az_span_size(ref_response->_internal.parser.remaining) == 0)
+  {
+    // avoid reading address if span is size 0
+    return AZ_ERROR_UNEXPECTED_END;
```

Comment:
This need to be documented in the header file as a possible return value for `az_http_response_get_next_header`